src/main/java/com/a5lab/tabr/controllers/TenantsController.java

```
@@ -32,41 +32,60 @@ public ModelAndView index() {
   }
 
-  @RequestMapping(value = "/settings/tenants/edit/{id}", method = RequestMethod.GET)
-  public ModelAndView edit(@PathVariable("id") long id) {
-    ModelAndView modelAndView = new ModelAndView("/settings/tenants/edit");
-    modelAndView.addObject("tenant", tenantService.findById(id));
-    return modelAndView;
+  @GetMapping(value = "/settings/tenants/edit/{id}")
+  public ModelAndView edit(@PathVariable("id") Long id, RedirectAttributes redirectAttributes) {
+    Optional<TenantRecord> tenantRecord = tenantService.findById(id);
+    if (tenantRecord.isPresent()) {
+      ModelAndView modelAndView = new ModelAndView("/settings/tenants/edit");
+      modelAndView.addObject("tenant", tenantRecord.get());
+      return modelAndView;
+    } else {
+      redirectAttributes.addFlashAttribute("msg_error", "Invalid tenant id. ");
+      return new ModelAndView("redirect:/settings/tenants/");
+    }
   }
 
-  @PostMapping("/settings/tenants/update")
-  public String update(@Valid TenantRecord tenant, BindingResult result) {
+  @PostMapping("/settings/tenants/update/{id}")
+  public String update(@PathVariable("id") Long id, @Valid TenantRecord tenantRecord,
+                       BindingResult result, RedirectAttributes redirectAttributes) {
     if (result.hasErrors()) {
-      return "/settings/tenants/add";
+      return "/settings/tenants/edit/{id}";
     }
-    tenantService.saveAndFlush(tenant);
+    tenantService.saveAndFlush(tenantRecord); // !!! a new insert? constraint failure
+    redirectAttributes.addFlashAttribute("msg_info", "The tenant has been updated successfully. ");
     return "redirect:/settings/tenants/";
   }
 
-  @RequestMapping(value = "/settings/tenants/{id}", method = RequestMethod.DELETE)
-  public String delete(@PathVariable("id") long id) {
+  @GetMapping(value = "/settings/tenants/delete/{id}")
+  public String delete(@PathVariable("id") Long id, RedirectAttributes redirectAttributes) {
     tenantService.deleteById(id);
```

You should check - if any row was deleted
Like SQL%rowcount anolog, every sql operation return how many rows were affected.
If no rows were deleted - riase an exception

---

@yurichervoniy 
here is signature void deleteById(ID id);
https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#deleteById(ID) "If the entity is not found in the persistence store it is silently ignored." return value is void. There are way to run: findById or just delete as is. My choice is the second one,

---

It depends on context and operation. 
It will be very strange when you try to delete a record - and get a success (silently ignored)
But it depends on....
Also i dont' like "get" for deletes, better to use post. But it's another story
Ok, let's use simple delete